lib/puppetlabs_spec_helper/rake_tasks.rb

```diff
@@ -500,10 +500,15 @@ def max_thread_limit
   end
 
   sha = `git rev-parse HEAD`[0..7]
+  branch = `git branch`
 
   # If we're in a CI environment include our build number
   if build = ENV['BUILD_NUMBER'] || ENV['TRAVIS_BUILD_NUMBER']
-    new_version = sprintf('%s-%04d-%s', version, build, sha)
+    if branch.include? "* release"
```

Comment:
I don't think there's any reason to make this generic? We only ever juggle the two branches on CI. If someone else adds another job they'll need to decide for themselves if it requires the amendment. Though I'll maybe add a comment saying why it's needed so others know what's going on here :)